modules/RTC/RTCUtils.js

```diff
@@ -555,6 +555,49 @@ function enumerateDevicesThroughMediaStreamTrack(callback) {
 }
 
 /**
+ * Returns a promise to check whether or not the permission prompt for user
+ * media might display. The promise returns a boolean but the boolean value is
+ * not a definitive indicator, except in the case of false as then it is certain
+ * no prompt will display.
+ *
+ * @param {Function} enumerateDevices - The native enumerate devices function.
+ * @returns {Promise<boolean>}
+ */
+function mightShowPermissionPrompt(enumerateDevices) {
+    // The focus check is necessary for browsers (Firefox) that do not complete
+    // getUserMedia when the browser tab is not in focus. In those cases, it is
+    // not known with certainty if the permission prompt will display or not.
+    if (!document.hasFocus()) {
```

Comment:
In this case enumerateDevices would finish and would be able to see labels. However, for firefo as long as the tab is not in focus the call to getusermedia will not complete. The workaround was put in place because the following will occur with the current implementation:
1. mightShowPermissionPrompt will return false.
2. This prevents PERMISSION_PROMPT_IS_SHOWN from firing, which prevents the permissions overlay from display. Only a gray screen is displayed.